src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/CSharpAssemblyDocumentGenerator.cs

```diff
@@ -0,0 +1,370 @@
+    private readonly bool _addPartialModifier;
+    private readonly bool _hideImplicitDefaultConstructors;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CSharpAssemblyDocumentGenerator"/> class.
+    /// </summary>
+    /// <param name="log">The logger to use.</param>
+    /// <param name="loader">The assembly symbol loader to use.</param>
+    /// <param name="symbolFilter">The symbol filter to use.</param>
+    /// <param name="attributeDataSymbolFilter">The attribute data symbol filter to use.</param>
+    /// <param name="exceptionMessage">The optional exception message to use.</param>
+    /// <param name="includeAssemblyAttributes">Whether to include assembly attributes or not.</param>
+    /// <param name="metadataReferences">The metadata references to use. The default value is <see langword="null"/>.</param>
+    /// <param name="addPartialModifier">Whether to add the partial modifier or not. The default value is <see langword="true"/>.</param>
+    /// <param name="hideImplicitDefaultConstructors">Whether to hide implicit default constructors or not. The default value is <see langword="true"/>.</param>
+    public CSharpAssemblyDocumentGenerator(ILog log,
+                                 IAssemblySymbolLoader loader,
+                                 ISymbolFilter symbolFilter,
+                                 ISymbolFilter attributeDataSymbolFilter,
+                                 string? exceptionMessage,
+                                 bool includeAssemblyAttributes,
+                                 IEnumerable<MetadataReference>? metadataReferences = null,
+                                 bool addPartialModifier = true,
+                                 bool hideImplicitDefaultConstructors = true)
+    {
+        _log = log;
+        _loader = loader;
+        _symbolFilter = symbolFilter;
+        _attributeDataSymbolFilter = attributeDataSymbolFilter;
+        _exceptionMessage = exceptionMessage;
+        _includeAssemblyAttributes = includeAssemblyAttributes;
+        _adhocWorkspace = new AdhocWorkspace();
+        _syntaxGenerator = SyntaxGenerator.GetGenerator(_adhocWorkspace, LanguageNames.CSharp);
+        _metadataReferences = metadataReferences;
+        _addPartialModifier = addPartialModifier;
+        _hideImplicitDefaultConstructors = hideImplicitDefaultConstructors;
+    }
+
+    /// <summary>
+    /// Returns the configured source code document for the specified assembly symbol.
+    /// </summary>
+    /// <param name="assemblySymbol">The assembly symbol that represents the loaded assembly.</param>
+    /// <returns>The source code document instance of the specified assembly symbol.</returns>
+    public Document GetDocumentForAssembly(IAssemblySymbol assemblySymbol)
+    {
+        CSharpCompilationOptions compilationOptions = new(OutputKind.DynamicallyLinkedLibrary,
+               nullableContextOptions: NullableContextOptions.Enable);
```

Comment:
Is there ever a time you will want this to not be enabled?

Reply:
I think both the ApiCompat and ApiDiff tools need to show nullability changes. I don't know if there's a case where this isn't needed.